File: FILE/TONGLP.py
with open("TONGLP.INP", "r") as f:
    n = int(f.readline())

with open("TONGLP.OUT", "w") as f:
    t = 0
    for i in range(1, n + 1):
        for j in range(i, n + 1):
            if (i ** 3 + j ** 3) == n:
                print(i, j)
                t += 1

File: CPP/TONGLP.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    freopen("TONGLP.INP", "r", stdin);
    freopen("TONGLP.OUT", "w", stdout);

    int n, t = 0;
    cin >> n;

    for (int i = 1; i <= n; i++) {
        for (int j = i; j <= n; j++) {
            if (pow(i, 3) + pow(j, 3) == n) {
                cout << i << " " << j << endl;
                t++;
            }
        }
    }
    cout << t;
}
